rpcs3/Emu/FS/vfsDirBase.h

```diff
@@ -62,14 +62,12 @@ class vfsDirBase
 
 	public:
 		iterator(vfsDirBase* parent)
-			: parent(parent)
-			, data(parent->First())
+			: data(parent->First())
 		{
 		}
 
 		iterator(const DirEntryInfo* data)
-			: parent(parent)
```

Comment:
Wouldn't parent be a nullptr in this case so ++ operator (with data = parent->Read();) would crash?

And on line 65 it seems to me that this shouldn't be deleted. Because there is something like this->parent(parent) .